src/main/java/reposense/authorship/FileInfoExtractor.java

```diff
@@ -223,4 +223,11 @@ private static int getStartingLineNumber(String linesChanged) {
 
         return Integer.parseInt(chunkHeaderMatcher.group(STARTING_LINE_NUMBER_GROUP_NAME));
     }
+
+    /**
+     * Returns true if {@code filePath} is valid and non-binary.
+     */
+    private static boolean isValidAndNonBinaryFile(String filePath, Set<Path> nonBinaryFilesSet) {
+        return (FileUtil.isValidPath(filePath) && nonBinaryFilesSet.contains(Paths.get(filePath)));
```

Comment:
Extra parenthesis `()`

Small optimization: It may be better to test this `nonBinaryFilesSet.contains(Paths.get(filePath)` first before testing `I/O`

Reply:
> Small optimization: It may be better to test this `nonBinaryFilesSet.contains(Paths.get(filePath)` first before testing `I/O`

Not possible in this current implementation, as attempting to parse paths with illegal windows characters into `nonBinaryFilesSet.contains(Paths.get(filePath)` will throw exceptions if we do not check for I/O first.

Reply:
Following https://github.com/reposense/RepoSense/commit/fed37581bbf17ae61a587256633be47924284b70, I think you can assume all file path will be valid?

Reply:
These two test cases do not call `GitLsTree.validateFilePaths` to check for the branch "windows illegal filename branch" as they directly checkout from master after validating that the master branch does not contain any illegal filenames:

https://github.com/reposense/RepoSense/blob/e1fbec2e4615302fabc345b5a0187358d72d8e29/src/test/java/reposense/authorship/FileInfoExtractorTest.java#L146-L157

https://github.com/reposense/RepoSense/blob/e1fbec2e4615302fabc345b5a0187358d72d8e29/src/test/java/reposense/authorship/FileInfoExtractorTest.java#L73-L85


Hence the need to check for valid file path

Reply:
This discussion is beyond this PR. If require, let's open a new issue for this.

My view is the unit test should be removed or changed to use `GitLsTree.validateFilePaths`. Now we are coding to accommodate this unit test. On runtime, we can assume that all filepaths are valid.

What do you think @eugenepeh?

Reply:
its good to have fail safe during run time to conform that assumption.
If the test is not thorough enough, then we better raise an issue for rework.